docs/developer/base/basics.md

```diff
@@ -1,3 +1,165 @@
+
+More information on this can be found in the API docs [here.](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck.rate)
+
+#### Histogram
+The histogram metric submission type represents the statistical distribution of a set of values calculated Agent-side in one time interval. Datadog’s histogram metric type is an extension of the StatsD timing metric type: the Agent aggregates the values that are sent in a defined time interval and produces different metrics which represent the set of values.
+
+More information on this can be found in the API docs [here.](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck.histogram)
+
+#### Historate
+Similar to the histogram metric, the historate represents statistical distribution over one time interval, although this is based on rate metrics.
+
+More information on this can be found in the API docs [here.](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck.historate)
+
+### Service Checks
+Service checks are a type of monitor used to track the uptime status of the service. For more information, see the [Service checks](https://docs.datadoghq.com/developers/service_checks/) guide.
+
+More information on this can be found in the API docs [here.](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck.service_check)
+
+### Events
+Events are informational messages about your system that are consumed by [the events stream](https://app.datadoghq.com/event/stream) so that you can build monitors on them.
+
+More information on this can be found in the API docs [here.](https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck.event)
+
+## Namespacing
+Within every integration, you can specify the value of `__NAMESPACE__`:
+```python
+from datadog_checks.base import AgentCheck
+
+
+class AwesomeCheck(AgentCheck):
+    __NAMESPACE__ = 'awesome'
+
+...
+```
+This is an optional addition, but it makes submissions easier since it prefixes every metric with the `__NAMESPACE__` automatically. In this case it would append `awesome.` to each metric submitted to Datadog.
+
+If you wish to ignore the namespace for any reason, you can append optional bool `raw=True` to each submission:
+```python
+...
+
+self.gauge('test', 1.23, tags=['foo:bar'], raw=True)
+
+...
+```
+We submitted a gauge metric named `awesome.test` with a value of `1.23` tagged by `foo:bar` ignoring the namespace.
+
+## Check Initializations
+In the AgentCheck class, there is a useful property called `check_initializations`, which we use to execute functions that are called once before the first check run.
```

Comment:
```suggestion
In the AgentCheck class, there is a useful property called `check_initializations`, which you can use to execute functions that are called once before the first check runs.
```

Reply:
Actually, it's correct as-is. This is referring to the first time a given check executes.

Reply:
I will word it to be less ambiguous with the word `run`.